### PR TITLE
LPS 184766 Inline Equinox Http Dev New

### PR DESCRIPTION
**Pull request by dantewang**
- LPS-184766 Create project files
- LPS-184766 Copy over Equinox Http Servlet source code (version 1.2.2.v20211118-0920.LIFERAY-PATCHED-1), no change
- LPS-184766 Move packages, no logic change
- LPS-184766 Add SF suppressions
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet and com.liferay.portal.osgi.web.http.servlet.context
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet.internal
- LPS-184766 Rename Activator to HttpServletBundleActivator and add it to bnd.bnd
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet.internal.context
- LPS-184766 SF and modify StringPlus to return String[] to fit most of its usages; this also prepares for the next commit.
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet.internal.customizer
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet.internal.error (unused classes are removed)
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet.internal.registration
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet.internal.servlet
- LPS-184766 SF, remove BooleanPlus and use GetterUtil instead
- LPS-184766 SF, remove ServiceProperties.parseBoolean() and use GetterUtil instead
- LPS-184766 Make SF happy: com.liferay.portal.osgi.web.http.servlet.internal.util (unused classes are removed)
- LPS-184766 SF, logging
- LPS-184766 SF, inline
- LPS-184766 SF, use HashMapDictionary and HashMapDictionaryBuilder
- LPS-184766 SF, use ReflectionUtil#throwException
- LPS-184766 SF, clean up
- LPS-184766 Apply, make portal-osgi-web-wab-extender depend on portal-osgi-web-http-servlet
- LPS-184766 Fix CI failures on JDK11: manually add imports that are in the original jar, but not automatically calculated when building the inlined module
- LPS-184766 Activator: Remove code that manually registers ProxyServlet; only register it via OSGi
- LPS-184766 Activator: Remove registered but unused OSGi service properties
- LPS-184766 Activator: Remove internal property, it's not part of the public standard nor used by Liferay
- LPS-184766 Activator: Simplify, track HttpServiceServlet directly
- LPS-184766 Activator: Simplify, use anonymous class
- LPS-184766 Activator: Change constant content
- LPS-184766 Activator: Move package, no logic change
- LPS-184766 Activator & Public Package: add a new interface to register the endpoint, register ProxyServlet internally, and remove HttpServiceServlet which extends the internal class
- LPS-184766 Activator & Public Package: move session listener registration into the HttpServletService tracker so that we don't need to expose HttpSessionTrackerUtil
- LPS-184766 Split into -api and -impl modules
- LPS-184766 Activator: Use HashMap(Dictionary)Builder to clean up
- LPS-184766 Activator: SF, inner class to buy space
- LPS-184766 Activator: SF, remove unused parameter in HttpTuple and inline in HttpServletImplBundleActivator
- LPS-184766 SF, remove unused class
- LPS-184766 HttpServiceRuntimeImpl: SF, inline
- LPS-184766 Activator: Use an alternative way to get the service.id set by the framework which is required by the Whiteboard spec, prepare for next
- LPS-184766 Activator & HttpTuple: Make HttpTuple to take an array of service registrations and unregister in reverse order when destroy
- LPS-184766 Activator & HttpServiceRuntimeImpl: Move default servlet context helper registration to the activator
- LPS-184766 SF, no need to be a list
- LPS-184766 SF, rename HttpTuple to HttpServiceBag and move
- LPS-184766 Clean up, remove interfaces that are not part of the whiteboard spec
- LPS-184766 HttpServiceRuntimeImpl: SF, split implementation of HttpServiceRuntime and other logic into 2 classes so that we don't pass an impl class everywhere.
- LPS-184766 HttpServiceRuntimeController: Use generic to simplify centralized Failed*DTO recording, and move other DTO logic to HttpServiceRuntimeImpl
- LPS-184766 HttpServiceRuntimeController: SF, move ServiceTrackerCustomizer logic inner class
- LPS-184766 HttpServiceRuntimeController: Move HttpService related logic to HttpServiceImpl, prepare for next
- LPS-184766 HttpServiceImpl: SF, use ConcurrentHashMap
- LPS-184766 HttpServiceImpl: No need to use map with Bundle as key. We registered a service factory instead of a single HttpServiceImpl instance, and each bundle will get its own instance of HttpServiceImpl, so it's naturally managed per bundle.
- LPS-184766 HttpServiceRuntimeController & ContextController: SF, remove unnecessary tracking and consuming bundle context as the support has been removed
- LPS-184766 Const: clean up constants, move or replace.
- LPS-184766 HttpServiceImpl: SF, inline HttpServiceObjectRegistration as inner class

### CODE REVIEW

**Comment by dantewang** [comment]
New Dev Branch:

Compared to https://github.com/dantewang/liferay-portal/pull/2305:

- Removed https://github.com/dantewang/liferay-portal/pull/2305/commits/b42a2a4b711eee0a4c6831e74c634ab6bb29c586 and Servlet 3.0 related part of https://github.com/dantewang/liferay-portal/pull/2305/commits/40666ee9bf850b9f5014d66a6434085bc5828db7
- Added "LPS-184766 HttpServiceImpl: SF, inline HttpServiceObjectRegistration as inner class`

cc @zhangyan0701 @tinatian

**Comment by dantewang** [comment]
@tinatian https://github.com/dantewang/liferay-portal/pull/2310/commits/36721fc6b5bd785383075025386c33c50298845d is the end of the last PR sent to BChan (and rejected).

**Comment by dantewang** [comment]
Force pushed to rebase, 2023-8-4 15:30

**Comment by dantewang** [comment]
ci:test

**Comment by dantewang** [comment]
ci:test

**Comment by tinatian** [comment]
@dantewang For https://github.com/dantewang/liferay-portal/pull/2310/commits/beff3b4bb6a75b8f3e28d5db38ab6b03b2a54c3b, why it is safe to remove?

**Comment by dantewang** [comment]
> @dantewang For https://github.com/dantewang/liferay-portal/pull/2310/commits/beff3b4bb6a75b8f3e28d5db38ab6b03b2a54c3b, why it is safe to remove?

@tinatian I think it's an Equinox specific feature, not part of any specification, and Liferay registers it via OSGi.

Also the Equinox Activator is an internal class, so the only way to trigger this method is to manually create an instance of the public sub-class of ProxyServlet. It means there's an API class depending on an impl class. I later replaced this with a new interface to be registered via OSGi, and removed the public sub-class of ProxyServlet.

**Comment by tinatian** [comment]
> > @dantewang For [beff3b4](https://github.com/dantewang/liferay-portal/commit/beff3b4bb6a75b8f3e28d5db38ab6b03b2a54c3b), why it is safe to remove?
> 
> @tinatian I think it's an Equinox specific feature, not part of any specification, and Liferay registers it via OSGi.
> 
> Also the Equinox Activator is an internal class, so the only way to trigger this method is to manually create an instance of the public sub-class of ProxyServlet. It means there's an API class depending on an impl class. I later replaced this with a new interface to be registered via OSGi, and removed the public sub-class of ProxyServlet.

@dantewang Please add this to commit message.

**Comment by tinatian** [comment]
@dantewang For this commit https://github.com/dantewang/liferay-portal/pull/2310/commits/6b2065d52f2e62f37e7c06e82c779f96ea6a8336, what are those interfaces for? is it safe to remove?

File: modules/apps/static/portal-osgi-web/portal-osgi-web-http-servlet-impl/src/main/java/com/liferay/portal/osgi/web/http/servlet/internal/util/Const.java [hunk 1]
```diff
@@ -16,43 +16,10 @@
...
-
-	public static final String LOCALHOST = "localhost";
-
-	public static final String OPEN_PAREN = "(";
-
-	public static final String SERVLET_NAME = "servlet-name";
-
-	public static final String SLASH = "/";
-
 	public static final String SLASH_STAR = "/*";
```
Comment (tinatian) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-http-servlet-impl/src/main/java/com/liferay/portal/osgi/web/http/servlet/internal/util/Const.java:19]: @dantewang Why we have something left in Const? Can you remove it?

Comment (dantewang) [reply modules/apps/static/portal-osgi-web/portal-osgi-web-http-servlet-impl/src/main/java/com/liferay/portal/osgi/web/http/servlet/internal/util/Const.java:19]: @tinatian We don't have such strings in our StringPool; and they are used in several places related to path matching. I will double check them, and move them to a more liferay-style place if I can't remove them.

**Comment by dantewang** [comment]
> @dantewang For this commit https://github.com/dantewang/liferay-portal/pull/2310/commits/6b2065d52f2e62f37e7c06e82c779f96ea6a8336, what are those interfaces for? is it safe to remove?

@tinatian This commit should be split into 2. 

For ExtendedHttpService, it's Equinox specific; the OSGi spec doesn't provide API to register Filter via HttpService.

For the context package, I just removed it because it's also Equinox API, and it's not used by Liferay. I didn't remember seeing anything related from the OSGi spec, but I'll double check.

**Comment by dantewang** [comment]
> > > @dantewang For [beff3b4](https://github.com/dantewang/liferay-portal/commit/beff3b4bb6a75b8f3e28d5db38ab6b03b2a54c3b), why it is safe to remove?
> > 
> > 
> > @tinatian I think it's an Equinox specific feature, not part of any specification, and Liferay registers it via OSGi.
> > Also the Equinox Activator is an internal class, so the only way to trigger this method is to manually create an instance of the public sub-class of ProxyServlet. It means there's an API class depending on an impl class. I later replaced this with a new interface to be registered via OSGi, and removed the public sub-class of ProxyServlet.
> 
> @dantewang Please add this to commit message.

@tinatian New PR with 5 new commits https://github.com/dantewang/liferay-portal/pull/2313

Commit message modified: https://github.com/dantewang/liferay-portal/pull/2313/commits/2e4c1d114c3ea9532ebfccbb0ee0375dc13d69f3

**Pull request closed without merging**